src/sim.rs

```diff
@@ -661,6 +686,207 @@ mod test {
+            .build();
+
+        let a_did_receive = Arc::new(Mutex::new(None));
+        let b_did_receive = Arc::new(Mutex::new(None));
+
+        let make_a = |sim: &mut Sim| {
+            sim.client(host_a, {
+                let a_did_receive = Arc::clone(&a_did_receive);
+                async move {
+                    let udp_socket =
+                        UdpSocket::bind((IpAddr::V4(Ipv4Addr::UNSPECIFIED), 1234)).await?;
+                    udp_socket
+                        .send_to(&[42], format!("{}:1234", host_b))
+                        .await
+                        .expect("sending packet should appear to work, even if partitioned");
+
+                    *a_did_receive.lock().unwrap() = Some(matches!(
+                        tokio::time::timeout(
+                            Duration::from_secs(1),
+                            udp_socket.recv_from(&mut [0])
+                        )
+                        .await,
+                        Ok(Ok(_))
+                    ));
+
+                    Ok(())
+                }
+            })
+        };
+
+        let make_b = |sim: &mut Sim| {
+            sim.client(host_b, {
+                let b_did_receive = Arc::clone(&b_did_receive);
+                async move {
+                    let udp_socket =
+                        UdpSocket::bind((IpAddr::V4(Ipv4Addr::UNSPECIFIED), 1234)).await?;
+                    udp_socket
+                        .send_to(&[42], format!("{}:1234", host_a))
+                        .await
+                        .expect("sending packet should work");
+
+                    *b_did_receive.lock().unwrap() = Some(matches!(
+                        tokio::time::timeout(
+                            Duration::from_secs(1),
+                            udp_socket.recv_from(&mut [0])
+                        )
+                        .await,
+                        Ok(Ok(_))
+                    ));
+
+                    Ok(())
+                }
+            })
+        };
+
+        let construct_a_first = sim.world.borrow_mut().rng.gen_bool(0.5);
+        if construct_a_first {
+            make_a(&mut sim);
+            make_b(&mut sim);
+        } else {
+            make_b(&mut sim);
+            make_a(&mut sim);
+        }
+
+        let Expectation {
+            expect_a_receive,
+            expect_b_receive,
+        } = partitioning(&mut sim);
+        sim.run()?;
+
+        assert_eq!(*a_did_receive.lock().unwrap(), Some(expect_a_receive));
+        assert_eq!(*b_did_receive.lock().unwrap(), Some(expect_b_receive));
+
+        Ok(())
+    }
+
+    #[test]
+    fn partition_peers_oneway() -> Result {
+        run_with_partitioning("a", "b", |sim: &mut Sim| {
+            sim.partition_oneway("a", "b");
+            Expectation {
+                expect_a_receive: true,
+                expect_b_receive: false,
+            }
+        })
+    }
+
+    #[test]
+    fn partition_peers_oneway_many_cases() -> Result {
```

Comment:
How's the duration on this now?

Reply:
It's quite fast.It's about 1/100 of the number of cases compared to before, which really improved performance.